### PR TITLE
🏗️ Build mixin + spiders: bcc (redux)

### PR DESCRIPTION
**Pull request by SimmonsRitchie**
## What's this PR do?
Adds a mixin and spiders to scrape multiple meetings from the website of Bismarck City Commission The mixin that handles the core scraping logic is called `bcc`.

Note: Unfortunately I hit the wrong button and accidentally merged the initial PR for these changes. I reverted that here #6. Hence "redux" in the PR title.

## Why are we doing this?
Requested by our site partners.

## Steps to manually test

1. Ensure the project is installed:
```
pipenv sync --dev
```
2. Activate the virtual env and enter the pipenv shell:
```
pipenv shell
```
3. Run the spiders:
```
scrapy list | grep 'bcc' | xargs -I {} scrapy crawl {} -O {}.csv
```
4. Monitor the output and ensure no errors are raised.

5. Inspect the CSVs to ensure the data looks valid.

## Are there any smells or added technical debt to note?
- Unfortunately the commit history got wiped here. I accidentally merged the initial PR (#5) before review. Refer to that PR if you wish to see the commit history.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206618610539948

### CODE REVIEW

File: tests/test_bisnd_bcc_a.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+from city_scrapers_core.constants import COMMITTEE, PASSED
+from city_scrapers_core.utils import file_response
+from freezegun import freeze_time
+
+from city_scrapers.spiders.bisnd_bcc import BisndBCCASpider
+
+# Setup for spider
+test_response = file_response(
+    join(dirname(__file__), "files", "bisnd_bcc_a.html"),
+    url="https://spokanevalley.granicus.com/ViewPublisher.php?view_id=3",
```
Comment (LienDang) [inline tests/test_bisnd_bcc_a.py]: URL from [spokanevalley.granicus.com](https://spokanevalley.granicus.com)?

**Comment by LienDang** [comment]
@SimmonsRitchie Aside from the misleading URL [spokanevalley.granicus.com](https://spokanevalley.granicus.com/) in the test file `tests/test_bisnd_bcc_a.py`, other files look good to me. So feel free to merge after updating the URL.

**Comment by SimmonsRitchie** [comment]
> @SimmonsRitchie Aside from the misleading URL [spokanevalley.granicus.com](https://spokanevalley.granicus.com/) in the test file `tests/test_bisnd_bcc_a.py`, other files look good to me. So feel free to merge after updating the URL.

Ah, thanks for catching that, @LienDang. Now fixed.

I've also fixed another issue I caught when I re-ran the spiders locally. For some reason, unlike when I tested this spider last week, all of the initial POST requests by these spiders were returning 404s. This behavior continued even after switching IP addresses with my VPN. Adding a `user-agent` appears to return 200 responses (at least for now). We may need to keep an eye on this mixin in prod. It's possible this site's administrators might take steps to limit scraping.

**Comment by LienDang** [comment]
Hi @SimmonsRitchie Recently while using an AWS account to access City Bureau's AWS resources, I discovered that there is a VPN server set up there in the EC2 service since 2019. I believe it is used to run the `scrapy` command, and there are configurations for it in the first city-scrapers repo https://github.com/City-Bureau/city-scrapers/blob/main/.github/workflows/cron.yml#L20-L22
My guess is that they built this OpenVPN server for the target site's admins to white list this IP address (there must have been an agreement between the 2 teams). If we follow this approach, we also need to figure a way to retrieve the existing credentials to set up them as secrets in the new repos, or produce another new credentials to access this server.

**Comment by SimmonsRitchie** [comment]
> Hi @SimmonsRitchie Recently while using an AWS account to access City Bureau's AWS resources, I discovered that there is a VPN server set up there in the EC2 service since 2019. I believe it is used to run the `scrapy` command, and there are configurations for it in the first city-scrapers repo https://github.com/City-Bureau/city-scrapers/blob/main/.github/workflows/cron.yml#L20-L22 My guess is that they built this OpenVPN server for the target site's admins to white list this IP address (there must have been an agreement between the 2 teams). If we follow this approach, we also need to figure a way to retrieve the existing credentials to set up them as secrets in the new repos, or produce another new credentials to access this server.

Yeah, I've noticed that some of our repos use this VPN server but many (maybe most?) do not. It's been a bit unclear to me how necessary this is. To be honest, I've been a bit reluctant to implement it in our new repos because I've noticed it can take quite a few minutes to kick into gear, which slows down how fast our production scrape does its business. I haven't explored why this is. My feeling is that it might just be better to enable this server on a case-by-case basis unless we learn something new that suggests we should implement it across all the repos? Thoughts?

**Comment by LienDang** [comment]
> Yeah, I've noticed that some of our repos use this VPN server but many (maybe most?) do not. It's been a bit unclear to me how necessary this is. To be honest, I've been a bit reluctant to implement it in our new repos because I've noticed it can take quite a few minutes to kick into gear, which slows down how fast our production scrape does its business. I haven't explored why this is. My feeling is that it might just be better to enable this server on a case-by-case basis unless we learn something new that suggests we should implement it across all the repos? Thoughts?

Agree with you about enabling the VPN server on a case base case basic when we need it. Just mentioned about this option (maybe you haven't noticed about it yet) so that it could be a solution for future cases.

**Review by LienDang — APPROVED** [review]
LGTM

**Pull request merged**